### PR TITLE
Blog App: Add Authorization Rules

### PR DESCRIPTION
**Pull request by DJ-MrJay**
In this milestone, we:
- Installed CanCanCan
- Added a role column to the `users` table using a migration
- Used CanCanCan to ensure that a user can delete a post if it is theirs or if they have an admin role
- Added the "Delete" button to the view and made sure that only authorized users can see it
- Used CanCanCan to ensure a user can delete a comment if it is theirs or if they have an admin role
- Added the "Delete" button to the view and made sure that only authorized users can see it

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,20 +1,37 @@
...
+    user = @comment.author_id
+    post = @comment.post_id
+
+    if @comment.destroy
+      flash[:success] = 'Comment was successfully deleted.'
+    else
+      flash[:error] = 'Error: Comment could not be deleted.'
+    end
+    redirect_to user_post_path(user, post)
+  end
```
Comment (AmaduKamara) [inline app/controllers/comments_controller.rb:33]: - [ ] In order to be able to destroy a comment, it needs to be created first and I was unable to create a comment for a user's post when posts are created. Kindly make it possible for comments to be created so the destroy feature can work as expected 👍🏽

I keep getting this error on the screenshot below when I try to create a comment for a post. 

Screenshot 2023-07-29 at 1 26 22 PM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @DJ-MrJay, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Used CanCan gem as required 👍🏽
✅ Set up CanCan Ability class 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] A user is unable to create a comment for a post or like them. Kindly work on these features so posts can be commented on and only then comments can be destroyed according to the requirement expectation 👍🏽

**Unable to create a comment for a post**

Screenshot 2023-07-29 at 1 26 22 PM

**Unable to Like a post**

Screenshot 2023-07-29 at 1 25 33 PM


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (DJ-MrJay) [reply app/controllers/comments_controller.rb:33]: Kindly call me on Zoom. Everything seems to be working on my end.

**Comment by AmaduKamara** [comment]
Hi @DJ-MrJay, thank you for reaching out on the issues I was having on my end.

It was good we connected on Zoom and saw the complete working demo of your app.

Kindly re-submit for a review so the next reviewer can approve your project. Your app is working fine as expected 👍🏽

**Review by algerina — APPROVED** [review]
Hi @DJ-MrJay  ,

Great job so far!  

May I highlight the following: 
- [x] Your  have a descriptive PR and commit history👏  ✔️ 
- [x] Your project seems to work well✔️ 

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**